cppcache/src/TcrConnection.cpp

```diff
@@ -592,56 +593,26 @@ Connector* TcrConnection::createConnection(const char* endpoint,
  *           that is used instead
  *   Body: default timeout
  */
-inline ConnErrType TcrConnection::receiveData(char* buffer, int32_t length,
-                                              uint32_t receiveTimeoutSec,
-                                              bool checkConnected,
-                                              bool isNotificationMessage,
-                                              int32_t notPublicApiWithTimeout) {
+inline ConnErrType TcrConnection::receiveData(
+    char* buffer, int32_t length, std::chrono::microseconds receiveTimeoutSec,
+    bool checkConnected, bool isNotificationMessage) {
   GF_DEV_ASSERT(buffer != nullptr);
   GF_DEV_ASSERT(m_conn != nullptr);
 
-  // if gfcpp property unit set then sendTimeoutSec will be in millisecond
-  // otherwise it will be in second
-  if (m_poolDM->getConnectionManager()
-          .getCacheImpl()
-          ->getDistributedSystem()
-          .getSystemProperties()
-          .readTimeoutUnitInMillis()) {
-    LOGFINER("recieveData %d %d ", receiveTimeoutSec, notPublicApiWithTimeout);
-    if (notPublicApiWithTimeout == TcrMessage::QUERY ||
```

Comment:
What happened to all this code? Was it not needed?

Reply:
Correct. All this code was figuring out if that “treat time as milliseconds” flag as set and if message was of any of these types then treat a milliseconds otherwise seconds. All goes away with strongly typed time units.

Reply:
Correct. All this code was figuring out if that “treat time as milliseconds” flag as set and if message was of any of these types then treat a milliseconds otherwise seconds. All goes away with strongly typed time units.